src/uct/tcp/tcp_cm.c

```diff
@@ -569,6 +575,11 @@ ucs_status_t uct_tcp_cm_conn_start(uct_tcp_ep_t *ep)
         }
     }
 
+    status = uct_tcp_cm_send_crc(ep);
```

Comment:
can't you send it in the  beginning of `CONN_REQ`?

Reply:
actually, not all `CONN_REQ` messages require sending a magic number prior, e.g. when a user creates TCP EP from UCT API call, UCT/TCP checks if it has a connection to the same peer - if yes, UCT/TCP just sends `CONN_REQ` to ensure `RX` capability will be set on a receiver's EP.
So, a receiver expects the packet with the magic number as the first message only in case of the connection was accepted, i.e. the first messages after `accept()` completed